tests/plugins/wysiwygarea/detached.js

```diff
@@ -0,0 +1,263 @@
+
+				wait();
+			} );
+		},
+
+		'test reattached editor parent restores editor data with observed default dom object': function() {
+			if ( !isSupportedEnvironment() ) {
+				assert.ignore();
+			}
+
+			bender.editorBot.create( {
+				startupData: startupData,
+				name: 'editor' + new Date().getTime()
+			}, function( bot ) {
+				var editorContainerParent = bot.editor.container.getParent(),
+					parentParent = editorContainerParent.getParent();
+
+				editorContainerParent.remove();
+				parentParent.append( editorContainerParent );
+
+				assertReattachedData( bot, startupData, 'Reattached editor parent should restore editor data. Observing default object.' );
+
+				wait();
+			} );
+		},
+
+		'test reattached editor restores editor data with editor parent observed': function() {
+			if ( !isSupportedEnvironment() ) {
+				assert.ignore();
+			}
+
+			var observableParent = CKEDITOR.document.getById( 'editorDetachableParent' );
+
+			bender.editorBot.create( {
+				name: 'editor1',
+				startupData: startupData,
+				config: {
+					observableParent: observableParent.$
+				}
+			}, function( bot ) {
+				var editorContainer = bot.editor.container;
+
+				editorContainer.remove();
+				observableParent.append( editorContainer );
+
+				assertReattachedData( bot, startupData, 'Reattached editor should restore data. Observing editor parent.' );
+
+				wait();
+			} );
+		},
+
+		'test reattached editor parent restores editor data with observed parent of detached element': function() {
+			if ( !isSupportedEnvironment() ) {
+				assert.ignore();
+			}
+
+			var observedParent = CKEDITOR.document.getById( 'editorObservedParent1' );
+
+			bender.editorBot.create( {
+				name: 'editor2',
+				startupData: startupData,
+				config: {
+					observableParent: observedParent.$
+				}
+			}, function( bot ) {
+				var editorContainerParent = bot.editor.container;
+
+				editorContainerParent.remove();
+				observedParent.append( editorContainerParent );
+
+				assertReattachedData( bot, startupData, 'Reattached editor parent should restore data. Observing parent of detached element.' );
+
+				wait();
+			} );
+		},
+
+		'test reattached editor with iframe content contains the same data with observed default dom object': function() {
+			if ( !isSupportedEnvironment() ) {
+				assert.ignore();
+			}
+
+			bender.editorBot.create( {
+				startupData: startupData,
+				name: 'editor3',
+				config: {
+					extraAllowedContent: 'iframe'
+				}
+			}, function( bot ) {
+				var editorContainer = bot.editor.container,
+					editorContainerParent = editorContainer.getParent();
+
+				editorContainer.remove();
+				editorContainerParent.append( editorContainer );
+
+				assertReattachedData( bot, startupData, 'Reattached editor should have the same data with iframe. Observing default object.' );
+
+				wait();
+			} );
+		},
+
+		'test reattached editor makes editor data empty with observed invalid DOM object': function() {
```

Comment:
This and following tests fail on IE8.

Reply:
It's funny - they failed for me only with opened dev console :thinking: And I can run them with dev console -> fail. Then close this console and they pass. So it doesn't look like cache :thinking:

Reply:
Strange. For me they fail even without console.